soda/scientific/soda/scientific/anomaly_detection/feedback_processor.py

```diff
@@ -0,0 +1,169 @@
+"""Handles user feedback consumption."""
+# extract the regularity
+# generate a "fake" custom regressor which:
+#   # captures the delta between the predicted and actual value,
+#   # reproduces is forward at captured regularity with some smoothing/decay around the point
+
+from datetime import date
+from typing import Any, Dict
+
+import pandas as pd
+from soda.common.logs import Logs
+
+FEEDBACK_REASONS = {
+    "expectedDailySeasonality": {
+        "internal_remap": "daily_seasonality",
+        "frequency_unit": "D",
+        "frequency_value": 1,
+    },
+    "expectedWeeklySeasonality": {
+        "internal_remap": "weekly_seasonality",
+        "frequency_unit": "W",
+        "frequency_value": 1,
+    },
+    "expectedMonthlySeasonality": {
+        "internal_remap": "monthly_seasonality",
+        "frequency_unit": "M",
+        "frequency_value": 1,
+    },
+    "expectedYearlySeasonality": {
+        "internal_remap": "yearly_seasonality",
+        "frequency_unit": "Y",
+        "frequency_value": 1,
+    },
+}
+
+
+WEEK_ANCHORED_OFFSETS = {
+    "Monday": "MON",
+    "Tuesday": "TUE",
+    "Wednesday": "WED",
+    "Thursday": "THU",
+    "Friday": "FRI",
+    "Saturday": "SAT",
+    "Sunday": "SUN",
+}
+
+
+class FeedbackProcessor:
```

Comment:
Again for this class, we can move `run` method to the top, and prepend private methods with `_`.

Reply:
I'll move the run indeed. I might skip pretending the private methods for now. I think it's not that crucial and I would only do it if we decide we want to 100% enforce this convention. What do you think?